tests/sqllogictests/suites/query/select.test

```diff
@@ -195,8 +195,37 @@ select * from t1 where a in (1, 2) order by a;
 1 2
 2 3
 
+# fix: https://github.com/datafuselabs/databend/issues/16143
```

Comment:
Will the test be converted to a subquery from inlist?  The default value is 1024.

```
 ("inlist_to_join_threshold", DefaultSettingValue {
                    value: UserSettingValue::UInt64(1024),
                    desc: "Set the threshold for converting IN list to JOIN.",
                    mode: SettingMode::Both,
                    range: Some(SettingRange::Numeric(0..=u64::MAX)),
                }),
```

Reply:
yes, we have `set inlist_to_join_threshold = 1;` on line 190.